Src/Plugins/Security/SecurePlaintextSecretsValidators/SEC101_008.AwsCredentialsValidator.cs

```diff
@@ -55,71 +60,122 @@ protected override IEnumerable<ValidationResult> IsValidStaticHelper(IDictionary
+                                    string iamUser = match.Value.String.Substring("User: ".Length, match.Value.String.Length - trimmedChars);
+                                    message = $"the compromised AWS identity is '{iamUser}";
+                                }
+
+                                return ValidationState.Authorized;
+                            }
+
+                            case "InvalidClientTokenId":
+                            case "SignatureDoesNotMatch":
+                            {
+                                return ValidationState.NoMatch;
+                            }
                         }
 
-                        return ValidationState.Authorized;
+                        return ValidationState.Unauthorized;
                     }
 
-                    case "InvalidClientTokenId":
-                    case "SignatureDoesNotMatch":
+                    default:
                     {
-                        return ValidationState.NoMatch;
+                        return ReturnUnexpectedResponseCode(ref message, response.StatusCode, id);
                     }
                 }
-
-                message = $"An unexpected exception was caught attempting to authenticate AWS id '{id}': {e.Message}";
-                return ValidationState.Unknown;
             }
             catch (Exception e)
             {
-                message = $"An unexpected exception was caught attempting to authentic AWS id '{id}': {e.Message}";
+                message = $"An unexpected exception was caught attempting to authenticate AWS id '{id}': {e.Message}";
                 return ValidationState.Unknown;
             }
-
-            return ValidationState.Authorized;
         }
 
-        private string BuildAuthorizedMessage(string id, GetAccountAuthorizationDetailsResponse response)
+        private string BuildAuthorizedMessage(string id, Stream responseStream)
         {
             var policyNames = new List<string>();
 
-            foreach (ManagedPolicyDetail policy in response.Policies)
+            using var reader = XmlReader.Create(responseStream);
+
+            if (reader.ReadToFollowing("Policies"))
             {
-                policyNames.Add(policy.PolicyName);
+                using XmlReader policyReader = reader.ReadSubtree();
+
+                while (policyReader.ReadToFollowing("member"))
+                {
+                    if (policyReader.ReadToFollowing("PolicyName"))
+                    {
+                        policyNames.Add(reader.ReadElementContentAsString());
+                    }
+                }
             }
 
             string policyNamesText = string.Join(", ", policyNames);
+
             return $"id '{id}' is authorized for role policies '{policyNamesText}'.";
         }
+
+        private void ReadErrorCodeMessage(Stream responseStream, out string code, out string message)
```

Comment:
>ReadErrorCodeMessage [](http://example.com/codeflow?start=21&length=20)

Example error response XML:
<ErrorResponse xmlns="[https://iam.amazonaws.com/doc/2010-05-08/](https://iam.amazonaws.com/doc/2010-05-08/)">
  <Error>
    <Type>Sender</Type>
    <Code>InvalidClientTokenId</Code>
    <Message>The security token included in the request is invalid.</Message>
  </Error>
  <RequestId>2c00c8f7-88b0-4155-bb41-460520e8c61e</RequestId>
</ErrorResponse>